internal/transport/http2_client.go

```diff
@@ -1282,96 +1283,73 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 				contentTypeErr = fmt.Sprintf("transport: received the unexpected content-type %q", hf.Value)
 				break
 			}
-			// TODO: do we want to propagate the whole content-type in the metadata,
-			// or come up with a way to just propagate the content-subtype if it was set?
-			// ie {"content-type": "application/grpc+proto"} or {"content-subtype": "proto"}
-			// in the metadata?
 			mdata[hf.Name] = append(mdata[hf.Name], hf.Value)
 			isGRPC = true
 		case "grpc-encoding":
 			s.recvCompress = hf.Value
 		case "grpc-status":
-			code, err := strconv.Atoi(hf.Value)
-			if err != nil {
-				grpcStatus = status.New(codes.Internal, fmt.Sprintf("transport: malformed grpc-status: %v", err))
-				break
-			}
-			rawStatusCode = codes.Code(uint32(code))
+			rawStatus = hf.Value
 		case "grpc-message":
 			grpcMessage = decodeGrpcMessage(hf.Value)
 		case "grpc-status-details-bin":
-			v, err := decodeBinHeader(hf.Value)
+			sg, err := decodeGRPCStatusDetails(hf.Value)
```

Comment:
It's a style nit, but we usually prefer to avoid the new local variable:
```go
var err error
statusGen, err = decodeGRPCStatusDetails(hf.Value)
...
```

Then you also don't need the `break`